src/main/java/com/aws/greengrass/logmanager/model/LogFile.java

```diff
@@ -110,6 +111,28 @@ private String readBytesToString() {
         return "";
     }
 
+    /**
+     * Get the hash of the logfile with target lines.
+     * @param chan a nyte channel that allow us to read the file
+     * @return the calculated hash value of the logfile, empty string if not enough lines for digest.
+     */
+    public String hashString(SeekableByteChannel chan) {
+        String fileHash = HASH_VALUE_OF_EMPTY_STRING;
+        try {
+            if (!this.exists()) {
+                return fileHash;
+            }
+            // String stringToHash = readBytesToString();
+            String stringToHash = readBytesToString(chan);
+            if (!stringToHash.isEmpty()) {
+                fileHash = calculate(stringToHash);
+            }
+        }  catch (NoSuchAlgorithmException e) {
+            logger.atError().cause(e).log("The digest algorithm is invalid");
+        }
+        return fileHash;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 120:133, 143:155.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.